### PR TITLE
feat(navigation): waitForNavigation/goto should not wait until respon…

### PR DESCRIPTION
**Pull request by pavelfeldman**
…se finished

### CODE REVIEW

File: test/network.spec.js [hunk 1]
```diff
@@ -258,21 +258,21 @@ module.exports.describe = function({testRunner, expect, MAC, WIN, FFOX, CHROMIUM
...
-      const requests = [];
-      page.on('requestfinished', request => requests.push(request));
-      await page.goto(server.EMPTY_PAGE);
-      expect(requests.length).toBe(1);
-      expect(requests[0].url()).toBe(server.EMPTY_PAGE);
-      expect(requests[0].response()).toBeTruthy();
-      expect(requests[0].frame() === page.mainFrame()).toBe(true);
-      expect(requests[0].frame().url()).toBe(server.EMPTY_PAGE);
+      const response = await page.goto(server.EMPTY_PAGE);
+      await response.finished();
```
Comment (aslushnikov) [inline test/network.spec.js]: not needed

**Review by aslushnikov — APPROVED** [review]

File: test/network.spec.js [hunk 2]
```diff
@@ -257,22 +257,24 @@ module.exports.describe = function({testRunner, expect, MAC, WIN, FFOX, CHROMIUM
...
-    it('Page.Events.RequestFinished', async({page, server}) => {
-      const requests = [];
-      page.on('requestfinished', request => requests.push(request));
-      await page.goto(server.EMPTY_PAGE);
-      expect(requests.length).toBe(1);
-      expect(requests[0].url()).toBe(server.EMPTY_PAGE);
-      expect(requests[0].response()).toBeTruthy();
-      expect(requests[0].frame() === page.mainFrame()).toBe(true);
-      expect(requests[0].frame().url()).toBe(server.EMPTY_PAGE);
+    fit('Page.Events.RequestFinished', async({page, server}) => {
```
Comment (JoelEinbinder) [inline test/network.spec.js]: fit

**Comment by JoelEinbinder** [comment]
What does this fix?

**Comment by pavelfeldman** [comment]
It prevents goto from hanging forever on pages with chunked encoding.

**Comment by JoelEinbinder** [comment]
> It prevents goto from hanging forever on pages with chunked encoding.

Can we add a test for this?

**Pull request merged**